builtin/mainmenu/init.lua

```diff
@@ -78,7 +78,31 @@ local function init_globals()
 		end,
 		-- Filter function
 		function(element, gameid)
-			return element.gameid == gameid
+			if element.gameid == gameid then
+				return true
+			end
+			local gameconfig = Settings(pkgmgr.find_by_gameid(gameid).path .. "/game.conf")
+			local aliases = (gameconfig:get("gameid_alias") or ""):split()
+			for _, alias in pairs(aliases) do
+				alias = alias:trim()
+				if element.gameid == alias then
+					return true
+				else
+					local el_id_len = #element.gameid
+					if el_id_len > 5 and element.gameid:sub(el_id_len - 4) == "_game" then
```

Comment:
This "trim off `_game` suffix" normalization logic appears at least thrice. Why not extract it to a local helper function (though with pattern matching this would be short enough that I wouldn't really mind); why store "un-normalized" gameids in the first place?

Side note: It seems you might have missed the case where both gameid and alias need to be normalized (both end in `_game`)? Comparing normalized values should automatically address that too.

Reply:
Haven't missed that case, it would be caught by the first (direct) comparison.

Refactoring this all anyway.